jobs/involuntary-dissolutions/involuntary_dissolutions.py

```diff
@@ -73,7 +78,55 @@ def shell_context():
     app.shell_context_processor(shell_context)
 
 
-def initiate_dissolution_process(app: Flask):  # pylint: disable=redefined-outer-name,too-many-locals
+def create_invountary_dissolution_filing(business_id: int):
+    """Create a filing entry to represent an involuntary dissolution filing."""
+    filing = Filing()
+    filing.business_id = business_id
+    filing._filing_type = CoreFiling.FilingTypes.DISSOLUTION  # pylint: disable=protected-access
+    filing._filing_sub_type = DissolutionTypes.INVOLUNTARY  # pylint: disable=protected-access
+    filing.save()
+
+    return filing
+
+
+async def put_filing_on_queue(filing_id: int, app: Flask, qsm: QueueService):
+    """Send queue message to filer to dissolve business."""
+    try:
+        subject = app.config['NATS_FILER_SUBJECT']
+        payload = {'filing': {'id': filing_id}}
+        await qsm.publish_json_to_subject(payload, subject)
+    except Exception as err:  # pylint: disable=broad-except # noqa F841;
+        # mark any failure for human review
+        capture_message(
+            f'Queue Error: Failed to place filing {filing_id} on Queue with error:{err}',
+            level='error'
+        )
+
+
+def mark_eligible_batches_completed():
+    """Mark batches completed if all of their associated batch_processings are compeleted."""
+    AliasBatchProcessing = aliased(BatchProcessing)  # pylint: disable=invalid-name # noqa N806
```

Comment:
Had to alias `BatchProcessing` to get this query to work. The sql query is a bit weird but it was the only way I could figure out how to select batches where their associated batch_processings entries have status COMPLETED or WITHDRAWN.

<img width="645" alt="Screenshot 2024-06-20 at 12 11 39 PM" src="https://github.com/bcgov/lear/assets/122323255/8cd5a91d-c3ef-4895-98f3-7c91ddb67b99">

Reply:
does something like this work?

```
batches = (
        db.session.query(Batch)
        .join(BatchProcessing, Batch.id == BatchProcessing.batch_id)
        .filter(Batch.batch_type == 'INVOLUNTARY_DISSOLUTION')
        .filter(Batch.status != 'COMPLETED')
        .group_by(Batch.id)
        .having(
            ~db.session.query(BatchProcessing.id)
            .filter(BatchProcessing.batch_id == Batch.id)
            .filter(BatchProcessing.status.notin_(['COMPLETED', 'WITHDRAWN']))
            .exists()
        )
        .all()
    )
```

Reply:
Unfortunately no, it doesn't. I get `Select statement '<sqlalchemy.sql.selectable.Select object at 0x107e74fd0>' returned no FROM clauses due to auto-correlation; specify correlate(<tables>) to control correlation manually`, which is the issue I encountered initially.

Reply:
Unfortunately no, it doesn't. I get `Select statement '<sqlalchemy.sql.selectable.Select object at 0x107e74fd0>' returned no FROM clauses due to auto-correlation; specify correlate(<tables>) to control correlation manually`, which is the issue I encountered initially.

Reply:
ok, resolving.